File: Data Structures & Algorithms/special-array-i/submission-0.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True

        even = True
        for i in nums:
            if i % 2 == 0 and even == False:
                return False
            elif i % 2 == 1 and even == True:
                return False
            if i % 2 == 0 and even == True:
                even = False
            elif i % 2 == 1 and even == False:
                even = True
            
        return True

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
         vector<int> output;
  unordered_map<int, int> numMap;
  vector<pair<int, int>> ns;

  for (auto x : nums) {
    numMap[x]++;
  }

  for (auto x : numMap) {
    ns.push_back(make_pair(x.second, x.first));
  }
  sort(ns.rbegin(), ns.rend());
  int i = 0;
  for (auto x : ns) {
    i++;
    output.push_back(x.second);
    if (i == k) {
      return output;
    }
  }
  return output;
 
    }
};
